### DISCUSSION TITLE
Are there any API changes after v0.5.0?

### DISCUSSION THREAD

**Discussion by XDTD** [go-webauthn/webauthn#135] [Q&A] [answered]
Hello! I'm from the Casdoor community. We recently encountered an issue when using Webauthn, and the detailed description is [here](https://github.com/casdoor/casdoor/issues/1706). We have temporarily solved the bug by rolling back Webauthn in go.mod to v0.5.0, but we still need to find out what caused this issue to use the latest version of Webauthn.

I would like to ask if there are any API changes after v0.5.0, especially between v0.5.0 and v0.6.0. In addition, based on my personal attempts, it seems that the issue occurred since [this commit](https://github.com/go-webauthn/webauthn/commit/6abd3517301412ba1f6a25c0d88ce59b22a463c6) . I would greatly appreciate any feedback you can provide on this matter.

**Comment by james-d-elliott** [comment, marked as answer]
Per the [release notes](https://github.com/go-webauthn/webauthn/releases/tag/v0.6.0) (and linked issue PR etc) there was a fix to one of the structs which previously was not using the specific Marshaller interface (ghosted). The specifics of the change (specifically with the challenge) are that if you're using the Base64 Standard encoding/decoding that it will fail, you need to ensure you're using the Base64 URL encoding (unpadded).

**Reply by james-d-elliott** [reply]
Side note, this brings the library to parity with the specification and the upcoming .toJSON() methods. I'd also recommend [@simplewebauthn/browser](https://www.npmjs.com/package/@simplewebauthn/browser).

**Reply by XDTD** [reply]
> Side note, this brings the library to parity with the specification and the upcoming .toJSON() methods. I'd also recommend [@simplewebauthn/browser](https://www.npmjs.com/package/@simplewebauthn/browser).

Thanks for your suggestion, I will look into it

**Reply by james-d-elliott** [reply]
No worries! Let me know if you have other questions/suggestions, now or in the future.
